nnf/__init__.py

```diff
@@ -998,6 +998,53 @@ def pair(node: NNF) -> NNF:
 
         return pair(sentence)
 
+    def project(self: T_NNF, names: 't.FrozenSet[Name]') -> 'NNF':
+        """Dual of :meth:``forget``: will forget all variables not given"""
+        return self.forget(self.vars() - names)
+
+    def forget_aux(self: T_NNF) -> 'NNF':
+        """Returns a theory that forgets all of the auxillary variables"""
+        aux_vars = frozenset({v for v in self.vars() if isinstance(v, Aux)})
+        return self.forget(aux_vars)
+
+    def forget(self: T_NNF, names: 't.FrozenSet[Name]') -> 'NNF':
```

Comment:
To make this (and `project`) a little easier to use, I'd change `names`'s annotation to `t.Iterable[Name]` and run it through `frozenset` before passing it into the implementing methods (which can keep requiring frozensets).

That would come in handy in the tests, so you could write:
```python
vars1 = list(sentence.vars())[:2]
```
As well as `forget_aux`:
```python
return self.forget(v for v in self.vars() if isinstance(v, Aux))
```